### PR TITLE
[Fix] Create Checkpoint

### LINKED ISSUE #69: Writers in test_faster_rcnn.py are mocked incorrectly
**Describe the Bug:**

A few folders such as `<MagicMock name='mock.logdir' id='140420520377936'>` will show up after running `test_faster_rcnn.py` unittests. This is due to `FasterRCNN` constructor tries to access `writer.logdir` which is already mocked using `writer = MagicMock()`. We have temporary fix to simply add `logdir` to the mocked writer object. This requires some cleanup of these tests to fix writer mocking. 

```
# test_faster_rcnn.py
writer = MagicMock()

# XXX This is just a hot fix to prevent a mysterious folder such as:
# <MagicMock name='mock.logdir' id='140420520377936'> showed up after
# running this test.
writer.logdir = tmp_name

estimator = FastreRCNN(... writer=writer...)
```

**Environment:**

- OS + version: Mac OS 10.5.5
- datasetinsights version: 0.2.0
- _Environment_: miniconda

### PR DESCRIPTION
**Pull request by Saurav-D**
# Peer Review Information

Fix for `_create_checkpoint` when `checkpoint_dir` is `None`.

# Pull Request Check List

- [ ] Added **tests** for changed code.
- [ ] Updated **documentation** for changed code.

### CODE REVIEW

File: tests/test_checkpoint.py [hunk 1]
```diff
@@ -130,3 +130,16 @@ def test_get_loader_raises_error():
     filepath = "some/wrong/path"
     with pytest.raises(ValueError, match=r"Given path:"):
         EstimatorCheckpoint._get_loader_from_path(filepath)
+
+
+def test_create_writer_when_checkpoint_dir_none():
```
Comment (BlairLee) [inline tests/test_checkpoint.py:115]: It's better to add another test for `checkpoint_dir.startswith(const.GCS_BASE_STR)` case.

**Review by BlairLee — DISMISSED** [review]

File: tests/test_checkpoint.py (hunk 1 above)
Comment (Saurav-D) [reply tests/test_checkpoint.py:115]: It was handled in a test above, but I have broken down it to individual test cases.

**Review by adason — DISMISSED** [review]

File: datasetinsights/io/checkpoint.py [hunk 2]
```diff
@@ -24,38 +24,44 @@ class EstimatorCheckpoint:
...
+
+        if checkpoint_dir is None:
+            checkpoint_dir = const.DEFAULT_CHECKPOINT_DIR
+            if not os.path.exists(checkpoint_dir):
+                os.makedirs(checkpoint_dir)
+
+        if checkpoint_dir.startswith(const.GCS_BASE_STR):
+            writer = GCSEstimatorWriter(checkpoint_dir, estimator_name)
+        elif os.path.isdir(checkpoint_dir):
+            writer = LocalEstimatorWriter(checkpoint_dir, estimator_name)
```
Comment (BlairLee) [inline datasetinsights/io/checkpoint.py:60]: Please also make sure `estimators/base.py` is updated. E.g. In `create_estimator` method
```
checkpointer = EstimatorCheckpoint(
        estimator_name=name, log_dir=checkpoint_dir, distributed=False,
    )
```
Should be:
```
checkpointer = EstimatorCheckpoint(
        estimator_name=name, checkpoint_dir=checkpoint_dir, distributed=False,
    )
```

Comment (Saurav-D) [reply datasetinsights/io/checkpoint.py:60]: Done.

**Review by adason — DISMISSED** [review]

**Review by adason — APPROVED** [review]

**Pull request merged**